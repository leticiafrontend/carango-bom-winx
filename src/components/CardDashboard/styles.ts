import { createStyles, makeStyles } from '@material-ui/core/styles'

export const cardStyle = makeStyles(() =>
  createStyles({
    root: {
      minWidth: 275,
    },
    title: {
      paddingBottom: 20,
      fontWeight: 500,
    },
    subtitle: {
      fontWeight: 400,
    },
  }),
)
