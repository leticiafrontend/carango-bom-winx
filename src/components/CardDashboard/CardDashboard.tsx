import React from 'react'

import { Card, CardContent, Typography } from '@material-ui/core'

import { cardStyle } from './styles'

interface Dash {
  montante: number
  nomeDaMarca: string
  quantidadeDeVeiculos: number
}

export const CardDashboard = ({
  montante,
  nomeDaMarca,
  quantidadeDeVeiculos,
}: Dash) => {
  const classes = cardStyle()

  return (
    <Card className={classes.root}>
      <CardContent>
        <Typography variant="h5" color="secondary">
          {nomeDaMarca}
        </Typography>
        <Typography variant="h6">
          {montante.toLocaleString('pt-br', {
            style: 'currency',
            currency: 'BRL',
          })}
        </Typography>
        <Typography variant="h6">Carros: {quantidadeDeVeiculos}</Typography>
      </CardContent>
    </Card>
  )
}
