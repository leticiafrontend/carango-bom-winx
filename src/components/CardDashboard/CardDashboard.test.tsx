import React from 'react'
import { render, screen } from '@testing-library/react'

import { CardDashboard } from './CardDashboard'

describe('CardDashboard component', () => {
  it('O snapshot do componente deve permanecer sempre o mesmo', () => {
    const { container } = render(
      <CardDashboard
        montante={1000}
        nomeDaMarca="Fiat"
        quantidadeDeVeiculos={9}
      />,
    )

    expect(container.firstChild).toMatchSnapshot()
  })
})
