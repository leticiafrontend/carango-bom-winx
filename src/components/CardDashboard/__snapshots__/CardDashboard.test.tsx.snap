// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardDashboard component O snapshot do componente deve permanecer sempre o mesmo 1`] = `
<div
  class="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    class="MuiCardContent-root"
  >
    <h5
      class="MuiTypography-root makeStyles-title-2 MuiTypography-h5 MuiTypography-colorSecondary"
    >
      Fiat
    </h5>
    <h6
      class="MuiTypography-root makeStyles-subtitle-3 MuiTypography-h6"
    >
      R$Â 1.000,00
    </h6>
    <h6
      class="MuiTypography-root makeStyles-subtitle-3 MuiTypography-h6"
    >
      Carros: 
      9
    </h6>
  </div>
</div>
`;
